src/components/Facility/Consultations/VentilatorTable.tsx

```diff
@@ -0,0 +1,124 @@
+import { formatDateTime } from "@/Utils/utils";
+import { DailyRoundsModel } from "../../Patient/models";
+import { useTranslation } from "react-i18next";
+
+type VentilatorTableProps = {
+  dailyRoundsList?: DailyRoundsModel[];
+};
+
+export default function VentilatorTable(props: VentilatorTableProps) {
+  const { t } = useTranslation();
+  const { dailyRoundsList } = props;
+  let sortedData: DailyRoundsModel[] = [];
+
+  const sortData = (data: DailyRoundsModel[]) => {
+    return data.sort(function (a, b) {
+      const ad = new Date(a.taken_at ?? Date.now());
+      const bd = new Date(b.taken_at ?? Date.now());
+      return ad > bd ? 1 : -1;
+    });
+  };
+
+  const VentilatorTableRow = ({
+    dailyRound,
+    start_date,
+    end_date,
+  }: {
+    dailyRound: DailyRoundsModel;
+    start_date: string;
+    end_date: string;
+  }) => {
+    return (
+      <tr className="text-center text-sm">
+        <td className="max-w-52 px-2 py-2">{start_date}</td>
+        <td className="max-w-52 px-2 py-2">{end_date}</td>
+        <td className="max-w-52 px-2 py-2">
+          {t(`RESPIRATORY_SUPPORT__${dailyRound?.ventilator_interface}`)}
+        </td>
+        <td className="max-w-52 px-2 py-2">
+          {dailyRound?.ventilator_interface == "INVASIVE" ||
+          dailyRound?.ventilator_interface == "NON_INVASIVE"
+            ? t(`VENTILATOR_MODE__${dailyRound?.ventilator_mode}`)
+            : dailyRound?.ventilator_interface == "OXYGEN_SUPPORT"
+              ? t(`OXYGEN_MODALITY__${dailyRound?.ventilator_oxygen_modality}`)
+              : null}
+        </td>
+      </tr>
+    );
+  };
+
+  const getModeOrModality = (round: DailyRoundsModel) => {
+    const ventilatorInterface = round.ventilator_interface;
+    if (!ventilatorInterface) return null;
+    const modeOrModality =
+      ventilatorInterface == "INVASIVE" || ventilatorInterface == "NON_INVASIVE"
+        ? round.ventilator_mode
+        : ventilatorInterface == "OXYGEN_SUPPORT"
+          ? round.ventilator_oxygen_modality
+          : null;
+    return modeOrModality;
+  };
+
+  const VentilatorTableBody = (dailyRoundsList: DailyRoundsModel[]) => {
+    const rows = [];
+    for (let index = 0; index < dailyRoundsList.length; index++) {
```

Comment:
a map should be more efficient and easy to use than a for loop here

Reply:
True, though within this loop, I'm traversing/skipping consecutive rounds (the while loop) that have same interface and mode/modality. That should be more readable with for loop, no? 🤔 

Same for below.

Reply:
Sorry just saw this. If it seems more readable to you this way sure